content_script.js

```diff
@@ -137,38 +140,104 @@ class HTMLSerializer {
    *
    * @param {Element} element The element being processed, which has the src
    *     attribute.
+   * @param {Set<string>} processedAttributes The Set containing all attributes
+   *     already added to |this.html| for the Element.
    * @private
    */
-  processSrcAttribute(element) {
-    if (element.tagName.toLowerCase() != 'iframe') {
-      this.processSrcHole(element);
-    } 
+  processSrcAttribute(element, processedAttributes) {
+    var tag = element.tagName;
+    switch (tag) {
+      case 'IFRAME':
+        break; // Do nothing.
+      case 'SOURCE':
+        if (!element.parent || element.parent.tagName != 'IMG') {
+          this.processSimpleSrc(element, processedAttributes);
```

Comment:
I restructured the code so that all of the processing for each case is explicitly handled.  Also, you are right about not being able to have an IMG for a parent.  IMGs can't have any children because they don't have closing tags.  I think what I meant to have was 'element.parent.tagName != 'PICTURE'.

Let me know what you think.